packages/core/src/utils.ts

```diff
@@ -243,3 +248,61 @@ export async function getRawConfiguration(
   }
   return sygmaConfig;
 }
+
+/**
+ * Validate Substrate address.
+ * @param {string} address
+ */
+export function isValidSubstrateAddress(address: string): boolean {
+  try {
+    encodeAddress(isHex(address) ? hexToU8a(address) : decodeAddress(address));
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
+/**
+ * Validate EVM address.
+ * @param {string} address
+ * @returns {boolean}
+ */
+export function isValidEvmAddress(address: string): boolean {
+  return !!ethers.utils.isAddress(address);
+}
+
+/**
+ * Validate Bitcoin address.
+ * @param {string} address
+ * @returns {boolean}
+ */
+export function isValidBitcoinAddress(address: string): boolean {
+  try {
+    bitcoin.address.toOutputScript(address, bitcoin.networks.bitcoin);
```

Comment:
Thank you for suggesting to check with different types. I ended up adding a library that can validate all types of addresses and also added the tests. Regarding the networks this library can check the address regardless of the network and it is more important that the address belongs to Bitcoin and is valid. Because I don't want to ask users to provide the Bitcoin network. However, we can do the following way to make a condition if sygma-sdk environment variable is equal to dev or test then bitcoin networks is also "test' otherwise it is equal to "prod"

Reply:
Updated with networks